### PR TITLE
Migrate `src/runtime/link-accounts-flow` to TypeScript

### PR DESCRIPTION
**Pull request by ChrisAntaki**

### CODE REVIEW

File: src/runtime/link-accounts-flow-test.js [hunk 1]
```diff
@@ -327,7 +327,7 @@ describes.realWin('LinkCompleteFlow', (env) => {
   [
     {
       description: 'should default index to 0',
-      activityResultData: {},
+      activityResultData: null,
```
Comment (ChrisAntaki) [inline src/runtime/link-accounts-flow-test.js:330]: Note: This expands the test's branch coverage

File: src/runtime/link-accounts-flow.ts [hunk 2]
```diff
@@ -266,57 +259,42 @@ export class LinkCompleteFlow {
...
   }
 
-  /**
-   * @param {!Object} result
-   * @return {!Promise<boolean>}
-   * @private
-   */
-  async handleLinkSaveResponse_(result) {
+  private async handleLinkSaveResponse_(
+    result: LinkCompleteResponse
```
Comment (justinchou-google) [inline src/runtime/link-accounts-flow.ts:296]: is this intended? should result be of type `{}`?

Comment (ChrisAntaki) [reply src/runtime/link-accounts-flow.ts]: I just improved the types

File: src/runtime/link-accounts-flow.ts [hunk 3]
```diff
@@ -248,14 +247,10 @@ export class LinkCompleteFlow {
     this.entitlementsManager_.setToastShown(true);
     this.entitlementsManager_.unblockNextNotification();
     this.entitlementsManager_.reset(success);
-    if (response && response['entitlements']) {
-      this.entitlementsManager_.pushNextEntitlements(response['entitlements']);
```
Comment (ChrisAntaki) [inline src/runtime/link-accounts-flow.ts:252]: Note: This seems like dead code

File: src/runtime/link-accounts-flow.ts (hunk 2 above)
Comment (ChrisAntaki) [reply src/runtime/link-accounts-flow.ts]: Now there are types for `LinkSaveResponse`, `LinkConfirmResponse`, and `LinkCompleteResponse` (a combination of the first two)

File: src/runtime/link-accounts-flow.ts (hunk 3 above)
Comment (ChrisAntaki) [reply src/runtime/link-accounts-flow.ts:252]: Yeah, I think this was a feature idea from 2018 that never got implemented

File: src/runtime/link-accounts-flow.ts (hunk 2 above)
Comment (justinchou-google) [reply src/runtime/link-accounts-flow.ts]: nice!

**Review by justinchou-google — APPROVED** [review]

**Pull request merged**